### PR TITLE
Add chinese stopwords filter

### PR DESCRIPTION
**Pull request by Anemone95**
Subject: Add Chinese stopwords filter

### Feature or Bugfix

- Feature

### Purpose
- add a filter for Chinese stop words

### Detail
- When generating a Chinese index for search, sphinx.search.zh.SearchChinese.py does not filter the stop words, which will cause the generated file (`searchindex.js`) too large.
- The solution is (1) Read the Chinese stop words from `sphinx/search/stopwords/chinese_stopwords.txt`——sphinx.search.zh.SearchChinese#stopwords, and (2) Filter the stopwords in  `sphinx.search.zh.SearchChinese#word_filter`.

### CODE REVIEW

**Comment by tk0miya** [comment]
Sorry for not responding long. But I can't review this because I can't read Chinese at all. Could you ask to review this to other Chinese sphinx users? I'll merge this if anybody will approve this.

File: sphinx/search/zh.py [hunk 1]
```diff
@@ -266,9 +279,9 @@ def stem(self, word):
         # if not stemmed, but would be too short after being stemmed
         # avoids some issues with acronyms
         should_not_be_stemmed = (
-            word in self.latin_terms and
-            len(word) >= 3 and
-            len(self.stemmer.stem(word.lower())) < 3
+                word in self.latin_terms and
+                len(word) >= 3 and
+                len(self.stemmer.stem(word.lower())) < 3
```
Comment (tk0miya) [inline sphinx/search/zh.py:284]: It seems these indentation changes are not needed.

File: sphinx/search/stopwords/chinese_stopwords.txt [hunk 2]
```diff
@@ -0,0 +1,2043 @@
+$
```
Comment (tk0miya) [inline sphinx/search/stopwords/chinese_stopwords.txt:1]: The words with their length equal to 1 are already filtered by `word_filter()` method. Are these entry really needed?

**Comment by AA-Turner** [comment]
Closing PR, author hasn't responded for over a year.

A

**Pull request closed without merging**